File: performance-testing/scripts/reset_lockout.js
const sqlite3 = require('../../backend/node_modules/sqlite3').verbose();
const path = require('path');

const dbPath = path.resolve(__dirname, '../../backend/database.sqlite');

const db = new sqlite3.Database(dbPath, (err) => {
  if (err) {
    console.error('[reset-lockout] Error opening database:', err.message);
    process.exit(1);
  }
});

db.run('UPDATE users SET login_attempts = 0, locked_until = NULL', function(err) {
  if (err) {
    console.error('[reset-lockout] Failed to reset lockout:', err.message);
    db.close();
    process.exit(1);
  }
  
  const timestamp = new Date().toISOString();
  console.log(`[reset-lockout] Cleared lockout on ${this.changes} users at ${timestamp}`);
  
  db.get('SELECT COUNT(*) AS locked_count FROM users WHERE locked_until IS NOT NULL', (verifyErr, row) => {
    if (verifyErr) {
      console.error('[reset-lockout] Verification query error:', verifyErr.message);
    } else {
      console.log(`[reset-lockout] Verification: active locked users = ${row.locked_count}`);
    }
    db.close();
  });
});

File: hw05-performance/test-plans/_reset_lockout.py
import sqlite3
from pathlib import Path

db_path = Path(r"c:\DiskD\HCMUS\Semester9\SoftwareTesting\Repo\eshop-sut\backend\database.sqlite")
sql_path = Path(r"c:\DiskD\HCMUS\Semester9\SoftwareTesting\SoftwareTesting-HW\HW5\23127271\test-plans\reset-lockout.sql")
db = sqlite3.connect(str(db_path))
db.executescript(sql_path.read_text(encoding="utf-8"))
row = db.execute(
    "SELECT COUNT(*), IFNULL(SUM(login_attempts),0), "
    "SUM(CASE WHEN locked_until IS NOT NULL THEN 1 ELSE 0 END) "
    "FROM users WHERE email LIKE 'tram%@eshop.com'"
).fetchone()
db.commit()
db.close()
print("tram_users", row[0], "attempts_sum", row[1], "locked", row[2])
